File: atp-tour/src/main/java/com/silab/atptour/service/StatisticsService.java
package com.silab.atptour.service;

import com.silab.atptour.entity.Match;
import com.silab.atptour.entity.Statistics;

/**
 * Represents a service containing all the logic for statistics data management
 * 
 * @author Lazar
 */
public interface StatisticsService {

    /**
     * Saves statistics data in the database
     * 
     * @param statistics A {@link Statistics} object to be saved
     * 
     * @return Saved {@link Statistics}
     */
    public Statistics saveStatistics(Statistics statistics);

    /**
     * Finds statistics in the database based on the match to which it refers
     * 
     * @param match A {@link Match} for which statistics need to be found
     * 
     * @return A {@link Statistics} for the given {@link Match}
     */
    public Statistics findStatistics(Match match);
}

File: atp-tour-angular/src/app/core/services/statistics.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Match } from 'src/app/models/match.model';
import { Statistics } from 'src/app/models/statistics.model';
import { environment } from 'src/environments/environment';

/**
 * Service for player data management
 * 
 * @author Lazar
 */
@Injectable({
  providedIn: 'root'
})
export class StatisticsService {

  private API_URL: string = environment.API_URL;

  /**
   * @constructor
   * 
   * @param {HttpClient} httpClient 
   */
  constructor(private httpClient: HttpClient) { }

  /**
   * Saves statistics to the database
   * 
   * @param {Statistics} statistics Statistics to be saved
   * 
   * @returns {Observable<Statistics>} Saved statistics
   */
  public saveStatistics(statistics: Statistics): Observable<Statistics> {
    return this.httpClient.put<Statistics>(this.API_URL + "/statistics", statistics);
  }

  /**
   * Finds match statistics from the database
   * 
   * @param {Match} match A match for which it is necessary to find statistics
   * 
   * @returns Observable<Statistics> Saved statistics
   */
  public findStatistics(match: Match): Observable<Statistics> {
    return this.httpClient.post<Statistics>(this.API_URL + "/statistics/find", match)
  }
}